lib/src/configuration.dart

```diff
@@ -307,3 +304,31 @@ class RealmSchema extends Iterable<SchemaObject> {
   @override
   SchemaObject elementAt(int index) => _schema.elementAt(index);
 }
+
+///The signature of a callback that will be invoked whenever a [SyncError] occurs for the synchronized Realm.
+///
+/// Client reset errors will not be reported through this callback as they are handled by [ClientResetHandler].
+class SyncErrorHandler {
+  /// The callback that handles the [SyncError].
+  void Function(SyncError code) callback;
+
+  /// Invokes the [SyncError] handling [callback].
+  void call(SyncError error) => callback(error);
+
+  /// Initializes a new instance of of [SyncErrorHandler].
+  SyncErrorHandler(this.callback);
+}
+
+/// The signature of a callback that will be invoked if a client reset error occurs for this [Realm].
```

Comment:
Again, why not just a `typedef SyncClientResetErrorHandler = void Function(SyncClientResetError code)`

Reply:
see comment above